zuul-core/src/test/java/com/netflix/netty/common/proxyprotocol/ElbProxyProtocolChannelHandlerTest.java

```diff
@@ -14,59 +14,103 @@
+import static org.junit.Assert.fail;
 import com.google.common.net.InetAddresses;
 import com.netflix.netty.common.SourceAddressChannelHandler;
-import com.netflix.netty.common.proxyprotocol.ElbProxyProtocolChannelHandler;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
 import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.haproxy.HAProxyMessage;
 import io.netty.handler.codec.haproxy.HAProxyProtocolVersion;
 import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-import java.nio.charset.StandardCharsets;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-
 @RunWith(JUnit4.class)
 public class ElbProxyProtocolChannelHandlerTest {
 
     @Test
     public void noProxy() {
-        ElbProxyProtocolChannelHandler handler = new ElbProxyProtocolChannelHandler(/* withProxyProtocol= */ false);
         EmbeddedChannel channel = new EmbeddedChannel();
-        handler.addProxyProtocol(channel.pipeline());
+        channel.pipeline().addLast(ElbProxyProtocolChannelHandler.NAME,
+                new ElbProxyProtocolChannelHandler(/* withProxyProtocol= */ false));
         ByteBuf buf = Unpooled.wrappedBuffer(
                 "PROXY TCP4 192.168.0.1 124.123.111.111 10008 443\r\n".getBytes(StandardCharsets.US_ASCII));
         channel.writeInbound(buf);
 
         Object dropped = channel.readInbound();
         assertEquals(dropped, buf);
 
-        // TODO(carl-mastrangelo): the handler should remove itself, but it currently doesn't.
-        assertNotNull(channel.pipeline().context(ElbProxyProtocolChannelHandler.NAME));
-        assertNull(channel.attr(ElbProxyProtocolChannelHandler.ATTR_HAPROXY_VERSION).get());
-        assertNull(channel.attr(ElbProxyProtocolChannelHandler.ATTR_HAPROXY_MESSAGE).get());
+        assertNull(channel.pipeline().context(ElbProxyProtocolChannelHandler.NAME));
+        assertNull(channel.pipeline().context("HAProxyMessageChannelHandler"));
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_VERSION).get());
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_MESSAGE).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDRESS).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_PORT).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_ADDRESS).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_PORT).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_REMOTE_ADDR).get());
+    }
 
+    @Test
+    public void passThrough_ProxyProtocolEnabled_nonProxyBytes() {
```

Comment:
I dont understand this test, it looks like those bytes actually are nonProxyBytes contrary to the name

Reply:
Correct.
`shouldPassThroughWhenProxyProtocolEnabledButReceivesNonProxyBytes` is what this is about.

Reply:
the bytes don't have a `PROXY` prefix. Added a comment to ease readability.